README.md

```diff
@@ -1,6 +1,533 @@
+    pub duration: Duration,
+}
+```
+
+### Metadata - API Methods
+
+```rust
+impl<R: StreamReader> MediaParser<R> {
+    /// Retrieve metadata for the media container
+    pub async fn metadata(&mut self) -> Result<MediaMetadata> {
+        // Implementation...
+    }
+}
+```
+
+### Metadata - Usage Example
+
+```rust
+use media_parser::{MediaParser, FileStreamReader};
+
+#[tokio::main]
+async fn main() -> media_parser::Result<()> {
+    let reader = FileStreamReader::new("video.mp4");
+    let parser = MediaParser::new(reader);
+    
+    let metadata = parser.metadata().await?;
+
+    println!("Title: {:?}", metadata.title);
+    println!("Artist: {:?}", metadata.artist);
+    println!("Album: {:?}", metadata.album);
+    println!("Copyright: {:?}", metadata.copyright);
+    println!("Duration: {:?}", metadata.duration);
+    
+    Ok(())
+}
+```
+
+## 2. Retrieve Tracks
+
+Get information about all video, audio, and subtitle tracks including codecs and dimensions.
+
+### Tracks - Types
+
+```rust
+/// Track types supported by the parser
+#[derive(Debug, Clone, PartialEq)]
+pub enum TrackType {
+    Video,
+    Audio,
+    Subtitle,
+}
+
+/// Media track information
+#[derive(Debug, Clone, PartialEq)]
+pub struct Track {
+    pub id: u32,
+    pub type: TrackType,
+    pub codec: String,
+    pub language: Option<String>,
+    pub frame_width: Option<u32>,
+    pub frame_height: Option<u32>,
+    pub frame_rate: Option<f64>,
+}
+```
+
+### Tracks - API Methods
+
+```rust
+impl<R: StreamReader> MediaParser<R> {
+    /// Retrieve list of all tracks in the container
+    pub async fn tracks(&mut self) -> Result<Vec<Track>> {
+        // Implementation...
+    }
+}
+```
+
+### Tracks - Usage Example
+
+```rust
+use media_parser::{MediaParser, FileStreamReader, TrackType};
+
+#[tokio::main]
+async fn main() -> media_parser::Result<()> {
+    let reader = FileStreamReader::new("video.mp4");
+    let parser = MediaParser::new(reader);
+    
+    let tracks = parser.tracks().await?;
+    
+    for track in &tracks {
+        println!("Track ID: {}", track.id);
+        println!("Type: {:?}", track.type);
+        println!("Codec: {}", track.codec);
+        println!("Language: {:?}", track.language);
+        
+        if track.type == TrackType::Video {
+            println!("Resolution: {}x{}", track.frame_width.unwrap_or(0), track.frame_height.unwrap_or(0));
+            println!("Frame Rate: {:?}", track.frame_rate);
+        }
+    }
+    
+    Ok(())
+}
+```
+
+## 3. Retrieve Subtitles
+
+Extract subtitle data with timing information, filtered by track ID, language,
+or first available track.
+
+### Subtitles - Types
+
+```rust
+/// Subtitle entry with timing information
+#[derive(Debug, Clone, PartialEq)]
+pub struct Subtitle {
+    pub id: u32,
+    pub start_time: Duration,
+    pub end_time: Duration,
+    pub text: String,
+}
+
+/// Query options for subtitle retrieval
+#[derive(Debug, Clone)]
+pub enum SubtitleQuery {
```

Comment:
This enum with the parameter is a nifty pattern I'm mentally tucking away for another day